recipes/yaml/CMakeLists.txt.patch

```diff
@@ -0,0 +1,25 @@
+diff --git CMakeLists.txt CMakeLists.txt
+index e84c28c..eab70b2 100644
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -3,6 +3,13 @@
+ cmake_minimum_required (VERSION 2.8)
+ project (yaml C)
+ 
++# Allow the developer to select if Dynamic or Static libraries are built
++OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" OFF)
++SET (LIB_TYPE STATIC)
++IF (BUILD_SHARED_LIBS)
++    SET (LIB_TYPE SHARED)
++ENDIF (BUILD_SHARED_LIBS)
++
+ set (YAML_VERSION_MAJOR 0)
+ set (YAML_VERSION_MINOR 1)
+ set (YAML_VERSION_PATCH 6)
+@@ -12,5 +19,4 @@ file (GLOB SRC src/*.c)
+ 
+ include_directories (include win32)
+ add_definitions (-DHAVE_CONFIG_H -DYAML_DECLARE_STATIC)
+-add_library (yaml STATIC ${SRC})
+-
++add_library (yaml ${LIB_TYPE} ${SRC})
```

Comment:
No, Windows has 2 different things both with .lib, and you need both of
them. One filename has to change to have both in one package.

On Thu, Apr 21, 2016, 20:04 jakirkham <notifications@github.com> wrote:

> In recipes/yaml/CMakeLists.txt.patch
> <https://github.com/conda-forge/staged-recipes/pull/397#discussion_r60678605>
> :
>
> > ++OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" OFF)
> > ++SET (LIB_TYPE STATIC)
> > ++IF (BUILD_SHARED_LIBS)
> > ++    SET (LIB_TYPE SHARED)
> > ++ENDIF (BUILD_SHARED_LIBS)
> > ++
> > + set (YAML_VERSION_MAJOR 0)
> > + set (YAML_VERSION_MINOR 1)
> > + set (YAML_VERSION_PATCH 6)
> > +@@ -12,5 +19,4 @@ file (GLOB SRC src/*.c)
> > +
> > + include_directories (include win32)
> > + add_definitions (-DHAVE_CONFIG_H -DYAML_DECLARE_STATIC)
> > +-add_library (yaml STATIC ${SRC})
> > +-
> > ++add_library (yaml ${LIB_TYPE} ${SRC})
>
> Would there be a way to have both libraries have the same name and only do
> one build? I initially tried to find something like this, but was unable to
> get anything like that to work.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/conda-forge/staged-recipes/pull/397/files/20da14e8df6aa1519d098c524871ea2191674708#r60678605>
>